File: 1046-MaxConsecutiveOnesIii/1046-MaxConsecutiveOnesIii.cpp
// Last updated: 16/10/2025, 04:25:15
//Video Link: https://www.youtube.com/watch?v=TAMLsfGiyOg

class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int start = 0;
        int maxCons1 = 0;
        int zeroCnt = 0;
        int n = nums.size();
        for(int end=0;end<n;end++){
            if(nums[end] == 0){
                zeroCnt++;
            }
            while(zeroCnt > k){
                if(nums[start] == 0){
                    zeroCnt--;
                }
                start++;
            }
            maxCons1 = max(maxCons1,end-start+1);
        }
        return maxCons1;
    }
};

File: 1046-MaxConsecutiveOnesIii/1046-MaxConsecutiveOnesIii.java
// Last updated: 01/11/2025, 09:17:44
class Solution {
    public int longestOnes(int[] nums, int k) {
        int n = nums.length;
        int i = 0;
        int j = 0;
        int cnt = 0;
        int flip = 0;
        while(j < n){
            if(nums[j] == 0){
                flip++;
            }
            if(flip > k){
                while(flip > k){
                    if(nums[i] == 0){
                        flip--;
                    }
                    i++;
                }
            }
            cnt = Math.max(cnt, j - i + 1);
            j++;
        }
        return cnt;
    }
}